src/methods/SANA.cpp

```diff
@@ -1283,7 +1293,27 @@ void SANA::performChange(int type) {
         currentScore     = newCurrentScore;
 		exposedEdgesNumer= newExposedEdgesNumer;
         squaredAligEdges = newSquaredAligEdges;
+        MS3Numer         = newMS3Numer;
+        if (needMS3)
+        {
+            // score is calculated with denom 1 step behind if we do this here I guess
+            unsigned oldTarDeg, newTarDeg;
```

Comment:
By "step" do you mean internal iteration (ie., a tiny fraction of a second) or meta-iterations at the multi-pairwise.sh level? If the latter, then yes there's no other option until we implement the whole thing in C++; if the former.... I don't see why that's the case?